include/hal/library/responder/csrlib.h

```diff
@@ -48,6 +48,60 @@ extern bool libspdm_gen_csr(uint32_t base_hash_algo, uint32_t base_asym_algo, bo
                             uint8_t *opaque_data, uint16_t opaque_data_length,
                             size_t *csr_len, uint8_t *csr_pointer,
                             bool is_device_cert_model);
+
+/**
+ * Gen CSR, which is uesd for SPDM 1.3
+ *
+ *   If the device need reset to gen csr, the all case is in the table.
+ *   | Overwrite | Req CSRTtrackingTag | Pending CSR | Reset |             Res Action       |
+ *   |-----------|---------------------|-------------|-------|------------------------------|
+ *   |    No     |          0          |      No     |   -   |          ResetRequired       |
+ *   |    No     |          0          |      Yes    |   -   |     ResetRequired or Busy    |
+ *   |    No     |        Non-0        |   No Match  |   -   |           Unexpected         |
+ *   |    No     |        Non-0        |     Match   | Before|             Busy             |
+ *   |    No     |        Non-0        |   No Match  | After |             CSR              |
+ *   |    Yes    |          0          |      No     |   -   |          ResetRequired       |
+ *   |    Yes    |          0          |      Yes    |   -   |          ResetRequired       |
+ *   |    Yes    |        Non-0        |      -      |   -   |             Invalid          |
+ *
+ * @param[in]      base_hash_algo        Indicates the hash algorithm.
+ * @param[in]      base_asym_algo        Indicates the signing algorithm.
+ * @param[in]      need_reset            If true, then device needs to be reset to complete the CSR.
+ *                                       If false, the device doesn`t need to be reset to complete the CSR.
+ *
+ * @param[in]      request                A pointer to the SPDM request data.
+ * @param[in]      request_size           The size of SPDM request data.
+ *
+ * @param[in]      requester_info         Requester info to generate the CSR.
+ * @param[in]      requester_info_length  The length of requester info.
+ *
+ * @param[in]      opaque_data            opaque data to generate the CSR.
+ * @param[in]      opaque_data_length     The length of opaque data.
+ *
+ * @param[in]      csr_len      For input, csr_len is the size of store CSR buffer.
+ *                              For output, csr_len is CSR len for DER format
+ * @param[in]      csr_pointer  On input, csr_pointer is buffer address to store CSR.
+ *                              On output, csr_pointer is address for stored CSR.
+ *                              The csr_pointer address will be changed.
+ *
+ * @param[in]       is_device_cert_model  If true, the cert chain is DeviceCert model.
```

Comment:
`is_device_cert_model` -> `req_cert_model`
This is also now a `uint8_t` since the Requester can specify `DeviceCert`, `AliasCert`, or `GenericCert`.

Reply:
Thanks. I think maybe we can update this parameter when we enable this feature.

Reply:
At least for the API definition, the feature is now. Otherwise we'll need a new API to convert the `bool` to `uint8_t`.

Reply:
Thanks. I have changed `is_device_cert_model` to `req_cert_model`.
But I just added simple logic in the l`ibspdm_gen_csr_ex`, which need enhance in the future.
https://github.com/DMTF/libspdm/blob/819b8513b25751fa14722a715b518cf6bcdbee24/os_stub/spdm_device_secret_lib_sample/lib.c#L995-L999